foundation/foundation-configuration-store/src/main/java/com/boozallen/aissemble/configuration/policy/PropertyRegenerationPolicyManager.java

```diff
@@ -0,0 +1,190 @@
+package com.boozallen.aissemble.configuration.policy;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Configuration::Store
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.configuration.policy.json.PropertyRegenerationPolicyInput;
+import com.boozallen.aissemble.configuration.policy.regeneration.PropertyRegenerationRule;
+import com.boozallen.aissemble.configuration.policy.regeneration.PropertyRegenerationStrategy;
+import com.boozallen.aissemble.configuration.policy.exception.PropertyRegenerationPolicyException;
+import com.boozallen.aissemble.configuration.store.PropertyKey;
+import com.boozallen.aissemble.core.policy.configuration.policy.DefaultPolicy;
+import com.boozallen.aissemble.core.policy.configuration.policy.Policy;
+import com.boozallen.aissemble.core.policy.configuration.policy.Target;
+import com.boozallen.aissemble.core.policy.configuration.policy.json.PolicyInput;
+import com.boozallen.aissemble.core.policy.configuration.policy.json.rule.PolicyRuleInput;
+import com.boozallen.aissemble.core.policy.configuration.policymanager.AbstractPolicyManager;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Class used for loading and configuring policies for determining when to regenerate a given {@link Property}.
+ */
+public class PropertyRegenerationPolicyManager extends AbstractPolicyManager {
+
+    private static final String TARGETS_REGEX = "/([^/]+)/([^/]+)";
+
+    private Set<PropertyRegenerationPolicy> propertyRegenerationPolicies;
+
+    /**
+     * Gets the property regeneration specific policies.
+     * @return map of policies
+     */
+    public Set<PropertyRegenerationPolicy> getPropertyRegenerationPolicies() {
+        if(this.propertyRegenerationPolicies == null) {
+            this.propertyRegenerationPolicies = new HashSet<>();
+            Map<String, Policy> policies = super.getPolicies();
+            if(policies != null) {
+                for (Map.Entry<String, Policy> entry : policies.entrySet()) {
+                    Policy value = entry.getValue();
+                    if(value instanceof PropertyRegenerationPolicy) {
+                        this.propertyRegenerationPolicies.add((PropertyRegenerationPolicy) value);
+                    }
+                }
+            }
+        }
```

Comment:
Q: so it seems like we lazily load the regeneration policies if the set is null, but would there ever be a case where the set of regen policies is not null, but it would still be worth checking to see if any regen policies need to be added? Or are we confident that the initial loading is sufficient

Reply:
Given how the policy manager is used in the `configLoader` the set would always be null, so it makes sense to remove this null check.